rotkehlchen/chain/ethereum/modules/the_graph/decoder.py

```diff
@@ -0,0 +1,179 @@
+from rotkehlchen.chain.ethereum.modules.the_graph.constants import (
+    CPT_THE_GRAPH,
+    THE_GRAPH_CPT_DETAILS,
+)
+from rotkehlchen.chain.ethereum.utils import token_normalized_value
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    ActionItem,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails, EventCategory
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_GRT
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, DecoderEventMappingType
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+CONTRACT_THE_GRAPH_STAKING = string_to_evm_address('0xF55041E37E12cD407ad00CE2910B8269B01263b9')
+TOPIC_TRANSFER = b'\xdd\xf2R\xad\x1b\xe2\xc8\x9bi\xc2\xb0h\xfc7\x8d\xaa\x95+\xa7\xf1c\xc4\xa1\x16(\xf5ZM\xf5#\xb3\xef'  # noqa: E501
+# example delegate() call: https://etherscan.io/tx/0x6ed3377db652151fb8e4794dd994a921a2d029ad317bd3f2a2916af239490fec
+TOPIC_STAKE_DELEGATED = b'\xcd\x03f\xdc\xe5$}\x87O\xfc`\xa7b\xaaz\xbb\xb8,\x16\x95\xbb\xb1q`\x9c\x1b\x88a\xe2y\xebs'  # noqa: E501
+# example undelegate() call: https://etherscan.io/tx/0x5ca5244868d9c0d8c30a1cad0feaf137bd28acd9c3f669a09a3a199fd75ad25a
+TOPIC_STAKE_DELEGATED_LOCKED = b'\x040\x18?\x84\xd9\xc4P#\x86\xd4\x99\xda\x80eC\xde\xe1\xd9\xde\x83\xc0\x8b\x01\xe3\x9am!\x16\xc4;%'  # noqa: E501
+# example withdrawDelegated() call: https://etherscan.io/tx/0x49307751de5ba4cf98fccbdd1ab8387fd60a7ce120800212c216bf0a6a04acfa
+TOPIC_STAKE_DELEGATED_WITHDRAWN = b'\x1b.w7\xe0C\xc5\xcf\x1bX|\xebM\xae\xb7\xae\x00\x14\x8b\x9b\xda\x8fy\xf1\t>\xea\xd0\x8f\x14\x19R'  # noqa: E501
+
+
+class TheGraphDecoder(DecoderInterface):
+    def counterparties(self) -> list[CounterpartyDetails]:
+        return [THE_GRAPH_CPT_DETAILS]
+
+    def possible_events(self) -> DecoderEventMappingType:
+        return {CPT_THE_GRAPH: {
+            HistoryEventType.STAKING: {
+                HistoryEventSubType.DEPOSIT_ASSET: EventCategory.DEPOSIT,
+                HistoryEventSubType.REMOVE_ASSET: EventCategory.WITHDRAW,
+            },
+            HistoryEventType.INFORMATIONAL: {
+                HistoryEventSubType.NONE: EventCategory.INFORMATIONAL,
+            },
+        }}
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            CONTRACT_THE_GRAPH_STAKING: (self._decode_delegator_staking,),
+        }
+
+    # processes events from The Graph's staking contract
+    def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
```

Comment:
Avoid simple comments when it's meant for functions or methods. Turn into docstring

```suggestion
    def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
         """Processes events from The Graph's staking contract"""
```